openrelik_worker_common/password_utils.py

```diff
@@ -0,0 +1,116 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import logging
+import os
+import shutil
+import subprocess
+import tempfile
+import threading
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+def bruteforce_password_hashes(password_hashes, tmp_dir, timeout=300, extra_args=""):
+    """Bruteforce password hashes using Hashcat or john.
+
+    Args:
+      password_hashes (list): Password hashes as strings.
+      tmp_dir (str): Path to use as a temporary directory
+      timeout (int): Number of seconds to run for before terminating the process.
+      extra_args (str): Any extra arguments to be passed to Hashcat.
+
+    Returns:
+      list: of tuples with hashes and plain text passwords.
+
+    Raises:
+      RuntimeError if execution failed.
+    """
+    logger.info("Starting password hash bruteforce")
+    password_hash_temp_file = tempfile.NamedTemporaryFile(delete=False, mode="w+", dir=tmp_dir)
+    password_hashes_file_path = password_hash_temp_file.name
+    password_hash_temp_file.write("\n".join(password_hashes))
+
+    pot_file = os.path.join((tmp_dir or tempfile.gettempdir()), "hashcat.pot")
+    password_list_file_path = "/openrelik/password.lst"
+    password_rules_file_path = "/openrelik/openrelik-password-cracking.rules"
```

Comment:
Who is responsible for creating these? Shouldn't these be passed in as parameters?

```suggestion
    password_list_file_path = "/openrelik/password.lst"
    password_rules_file_path = "/openrelik/openrelik-password-cracking.rules"
```